docs/architecture.rst

```diff
@@ -0,0 +1,47 @@
+============
+Architecture
+============
+
+The project is build around the core library that is called ``guides``. This library
+contains all core components of the project. Like the different layers :doc:`parser`,
+:doc:`compiler` and :doc:`render` and includes the basic Nodes and templates to create
```

Comment:
should it say "renderer"? Or should every layer be followed by "component"?
- parser component
- compiler component
- render component

?

Reply:
each of them are file names, not what is displayed here. Text in the references will be taken from the pages titles.